scripts/gendoxylayout.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+# This python program generates parameters for the linker script generator feature.
+
+# This file is part of the libopencm3 project.
+#
+# 2019 Guillaume Revaillot <g.revaillot@gmail.com>
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library. If not, see <http://www.gnu.org/licenses/>.
+
+from lxml import etree
```

Comment:
lxml is a non-core library right?  and it's one that has to be compiled?

Reply:
if I'm reading right, the expat binding is the only one standard.  https://docs.python.org/2/library/xml.html (and same for v3)  If we use that, it's less deps to install.  I like the idea of this PR, but I don't really like adding deps for this

Reply:
arf, It looked python standard here - I'll have a look at that.